### PR TITLE
Master mppalves

### PR DESCRIPTION
**Pull request by mppalves**
Added new PastureSuit function (still need to implement 67k cells)

### CODE REVIEW

**Review by tscheypidi — APPROVED** [review]
from rse perspective the changes look good to me, but I cannot judge the changes from a content perspective. It would be important to hear from @k4rst3ns whether the changes are in line with the other functions. Please wait for her approval before merging.

File: R/fullCELLULARMAGPIE.R
Comment (k4rst3ns) [inline R/fullCELLULARMAGPIE.R:209]: climatetype and lpjml argument has to be set to the flexible arguments as they change with the preprocessing settings.

See e.g.: https://github.com/pik-piam/mrmagpie/blob/5b6edcabdcc51e52a944586258d038abaea9a3f6/R/fullCELLULARMAGPIE.R#L322

My guess how the call should look like:
 calcOutput("PastureSuit", climatetype = climatetype, lpjml =  lpjml[["natveg"]],
             file = paste0("f31_pastr_suitability_", ctype, ".mz"), years =  lpjYears, aggregate = "cluster")
             
Check which years make most sense (magYears or lpjYears) and the lpjml version ('natveg' or 'grass').

**Review by k4rst3ns — CHANGES_REQUESTED** [review]
This looks fine to me.

One thing I noticed:
You call PastureSuit in `calcPastrTauHist.R` but it is never used in there. https://github.com/pik-piam/mrmagpie/blob/5b6edcabdcc51e52a944586258d038abaea9a3f6/R/calcPastrTauHist.R#L23
Please remove the call there, as this calculation is rather costly and maybe not needed at that spot.

I couldn't find another spot except for the fullCellularmagpie, where calcPastureSuit is called, nevertheless it might makes sense to add the same handling of the climatetype argument as in calcYieldsLPJmL:
https://github.com/pik-piam/mrland/blob/beaafd170ff2989a7e0917a9a23ce40fccf84de7/R/calcYieldsLPJmL.R#L27
... that helps to set the correct stage depending on the climatetype setting (and if historical climatetype is used stage is set "only" to smoothed). the stage argument than has to be set in the mpet and precipitation.

File: R/calcPastureSuit.R
Comment (k4rst3ns) [inline R/calcPastureSuit.R:43]: It seems you only need yearly precipitation and pet, is this correct?
We might than already call the function for yearly sums as this would speed up harmonization, but might smooth out more extreme yearly fluctuations. Maybe we ask Jens here once again, what makes sense.

**Review by k4rst3ns — COMMENTED** [review]
sorry also added another comment ^^'

File: R/calcPastureSuit.R
Comment (k4rst3ns) [inline R/calcPastureSuit.R:50]: Do you really need cellsize or landarea size of a given cell? 
If you need this to calculate out of total numbers density - maybe landarea size per cell makes more sense?

there is a function already calculating this:
calcOutput("CellArea", aggregate = FALSE)

**Review by k4rst3ns — COMMENTED** [review]
Last comment:
PastureSuit is now flexibel with climate and lpjml version.
For GrasslandYields this is still to be done.

**Pull request merged**
